msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md

```diff
@@ -196,17 +196,25 @@ app.messageExtensions.selectItem(async (context: TurnContext, state: TurnState,
 
 # [Python](#tab/python5)
 
-* [Code sample](https://github.com/microsoft/teams-ai/tree/main/python/samples/04.ai.b.messageExtensions.AI-ME)
+* [Code sample](https://github.com/microsoft/teams-ai/tree/main/python/samples/02.messageExtensions.a.searchCommand)
 
-* [Sample code reference](https://github.com/microsoft/teams-ai/blob/main/python/samples/04.ai.b.messageExtensions.AI-ME/src/bot.py#L75)
+* [Sample code reference](https://github.com/microsoft/teams-ai/blob/main/python/samples/02.messageExtensions.a.searchCommand/src/bot.py#L44)
 
 ```python
-# Implement Message Extension logic
-@app.message_extensions.fetch_task("CreatePost")
-async def create_post(context: TurnContext, _state: AppTurnState) -> TaskModuleTaskInfo:
-    # Return card as a TaskInfo object
-    card = create_initial_view()
-    return create_task_info(card)
+@app.message_extensions.query("searchCmd")
+async def search_command(
+    _context: TurnContext, _state: AppTurnState, query: MessagingExtensionQuery
+) -> MessagingExtensionResult:
+    query_dict = query.as_dict()
+    search_query = ""
+    if query_dict["parameters"] is not None and len(query_dict["parameters"]) > 0:
+        for parameter in query_dict["parameters"]:
+            if parameter["name"] == "queryText":
+                search_query = parameter["value"]
+                break
+    count = query_dict["query_options"]["count"] if query_dict["query_options"]["count"] else 10
+    url = "http://registry.npmjs.com/-/v1/search?"
+    params = {"size": count, "text": search_query}
```

Comment:
missing extra code (need to click "Show all lines" in Loop component)

Reply:
Updated the complete code snippet.